src/server/pfs/server/testing/server_test.go

```diff
@@ -4698,6 +4698,37 @@ func TestPFS(suite *testing.T) {
 		require.ElementsEqualUnderFn(t, expectedCommits, commitInfos, CommitInfoToID)
 	})
 
+	suite.Run("SquashCommitEmptyChild", func(t *testing.T) {
+		t.Parallel()
+		env := testpachd.NewRealEnv(t, tu.NewTestDBConfig(t))
+
+		repo := "repo"
+		file := "foo"
+		require.NoError(t, env.PachClient.CreateRepo(repo))
+
+		commitA, err := env.PachClient.StartCommit(repo, "master")
+		require.NoError(t, err)
+		require.NoError(t, env.PachClient.PutFile(commitA, file, strings.NewReader("foo")))
+		require.NoError(t, env.PachClient.FinishCommit(repo, "master", ""))
+
+		commitB, err := env.PachClient.StartCommit(repo, "master")
+		require.NoError(t, err)
+
+		// squash fails, child is still open
+		err = env.PachClient.SquashCommitSet(commitA.ID)
+		require.YesError(t, err)
+		require.Matches(t, "finished", err.Error())
+
+		require.NoError(t, env.PachClient.FinishCommit(repo, "master", ""))
+		require.NoErrorWithinTRetry(t, 30*time.Second, func() error {
```

Comment:
This method is responsible for a lot of the CI timeouts with no useful information.  Can we instead call inspect commit and wait until it's finished?

Reply:
Yes, that's much better, will change